x-pack/legacy/plugins/maps/public/reducers/map.js

```diff
@@ -377,33 +380,38 @@ function updateSourceDataRequest(state, action) {
 }
 
 
+function clearDataRequest(state, action) {
```

Comment:
Introduced this new helper `clearDataRequest` to remove the data.

Isolated `findDataRequestByLayerId` since it is reused in two locations.